pytorch_lightning/core/lightning.py

```diff
@@ -1893,7 +1893,10 @@ def to_onnx(self, file_path: Union[str, Path], input_sample: Optional[Any] = Non
 
         if "example_outputs" not in kwargs:
             self.eval()
-            kwargs["example_outputs"] = self(input_sample)
+            if not isinstance(input_sample, Tuple):
+                kwargs["example_outputs"] = self(input_sample)
+            else:
+                kwargs["example_outputs"] = self(*input_sample)
```

Comment:
```suggestion
            if isinstance(input_sample, Tuple):
                kwargs["example_outputs"] = self(*input_sample)
            else:
                kwargs["example_outputs"] = self(input_sample)
```
does onnx also support kwargs? if so, we can have a third case and pass in `**input_sample`

Reply:
well, this is not about onnx, but about LightningModule/torch.nn.module. Per docs https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.core.lightning.html#pytorch_lightning.core.lightning.LightningModule.example_input_array the input may be Tuple, Dict or Tensor, however I haven't found any model in the repo having the example_input_array as Dict.

Reply:
this was precisely my question, whether onnx supports tracing with the kwargs. 
Since the convention of how args are passed in do not match here with Lightning, staying with your approach seems the best.